src/services/admin/mentor.service.ts

```diff
@@ -109,3 +109,52 @@ export const findAllMentorEmails = async (
     throw new Error('Error getting mentors emails')
   }
 }
+
+export const searchMentorsByCategory = async (
+  category?: string
+): Promise<{
+  statusCode: number
+  mentors?: Mentor[] | null
+  message: string
+}> => {
+  try {
+    const mentorRepository = dataSource.getRepository(Mentor)
+
+    const categoryQuery = category ? `${category}%` : ''
+
+    const mentors: Mentor[] = await mentorRepository
+      .createQueryBuilder('mentor')
+      .innerJoinAndSelect('mentor.category', 'category')
+      .leftJoinAndSelect('mentor.profile', 'profile')
+      .select([
+        'mentor.uuid',
+        'category.category',
+        'mentor.application',
+        'profile.contact_email',
+        'profile.first_name',
+        'profile.last_name',
+        'profile.image_url',
+        'profile.linkedin_url'
+      ])
```

Comment:
Without selecting fields, what does it return? @sathudeva7  Also can you drop a comment on the issue and get assigned?

Reply:
sorry for the mistake could you please assign me to this issue

Reply:
without selecting fields ->

![Screenshot 2023-10-24 at 16 27 54](https://github.com/sef-global/scholarx-backend/assets/67930416/f735c16c-543a-4959-a29a-010d0d21134f)

Reply:
@sathudeva7 Can you drop a comment on the issue then I will assign you

Reply:
@sathudeva7 Can you update it to without selecting?

Reply:
i will update within this week

Reply:
Any updates @sathudeva7?